### PR TITLE
Create Database Access screen with stories

### PR DESCRIPTION
**Pull request by rudream**
### Database Access Screen

Includes:

- Table list showing data of each database
- Connect button on each row that opens a dialog with dynamically generated instructions on how to connect to the selected database
- Add database button that opens a dialog with dynamically generated instructions on how to add a database
- Stories and tests

Enterprise PR for registering the feature and Main.story can be found [here](https://github.com/gravitational/webapps.e/pull/52/)

### CODE REVIEW

File: packages/teleport/src/Databases/ConnectDialog/ConnectDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+          {`* Note: To connect with a GUI database client, see our `}
+          <Link
+            href={
+              'https://goteleport.com/docs/database-access/guides/gui-clients/'
+            }
+            target="_blank"
+          >
+            documentation
+          </Link>
+          {` for instructions `}
```
Comment (kimlisa) [inline packages/teleport/src/Databases/ConnectDialog/ConnectDialog.tsx:90]:
```suggestion
          {` for instructions.`}
```
needs a period after instructions

File: packages/teleport/src/Databases/AddDialog/Manually/Manually.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        <Box>
+          {`* Note: For a self-hosted Teleport version, you may need to update DNS and obtain a TLS certificate for this application.
+            Learn more about database access `}
+          <Link
+            href={'https://goteleport.com/docs/database-access/'}
+            target="_blank"
+            mr="2"
+          >
+            here
+          </Link>
```
Comment (kimlisa) [inline packages/teleport/src/Databases/AddDialog/Manually/Manually.tsx:121]:
```suggestion
          <Link
            href={'https://goteleport.com/docs/database-access/'}
            target="_blank"
          >
            here
          </Link>
          .
```
need period after `here`

**Review by kimlisa — COMMENTED** [review]
language fixes

**Comment by kimlisa** [comment]
@rudream it looks like this PR includes the latest commit i made to master (session end events). I would've advised you to rebase with master, but considering you have one commit, you might need to do some reflogging to backtrack some commits: https://github.blog/2015-06-08-how-to-undo-almost-anything-with-git/

File: packages/teleport/src/Databases/AddDialog/AddDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        <Box mb={4}>
+          <Text bold as="span">
+            Step 2
+          </Text>
+          {' - Login to Teleport'}
+          <TextSelectCopy
+            mt="2"
+            text={`tsh login --proxy=${host} ${
+              authType === 'local' ? `--auth=local --user=${user}` : ''
+            }`}
```
Comment (kimlisa) [inline packages/teleport/src/Databases/AddDialog/AddDialog.tsx:92]: this is a little hard to read, i'd put it into a variable

File: packages/teleport/src/Databases/AddDialog/AddDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+import * as links from 'teleport/services/links';
+import { AuthType } from 'teleport/services/user';
+import { DbType, DbProtocol } from 'teleport/services/databases';
+import {
+  formatDatabaseInfo,
+  DatabaseInfo,
+} from 'teleport/services/databases/makeDatabase';
+import TextSelectCopy from 'teleport/components/TextSelectCopy';
+
+export default function AddDialog({ user, version, authType, onClose }: Props) {
```
Comment (alex-kovoy) [inline packages/teleport/src/Databases/AddDialog/AddDialog.tsx:35]: Lets rename it to `AddDatabase`. It's easier to search and less generic than "AddDialog". Another benefit is the name is implementation agnostic meaning that if we decide to use something else (not the dialog) we wont need to change the component name.

File: packages/teleport/src/Databases/ButtonAdd/ButtonAdd.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { ButtonPrimary } from 'design';
+
+const docUrl = 'https://goteleport.com/docs/database-access';
```
Comment (alex-kovoy) [inline packages/teleport/src/Databases/ButtonAdd/ButtonAdd.tsx:20]: `const DOC_URL = ` as this is a package level constant

File: packages/teleport/src/Databases/DatabaseList/DatabaseList.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+import ConnectDialog from 'teleport/Databases/ConnectDialog';
+
+function DatabaseList(props: Props) {
+  const { databases = [], pageSize = 20, user, clusterId, authType } = props;
+
+  const [sortDir, setSortDir] = useState<Record<string, string>>({
+    name: SortTypes.DESC,
+  });
+
+  const [searchValue, setSearchValue] = useState<string>('');
```
Comment (alex-kovoy) [inline packages/teleport/src/Databases/DatabaseList/DatabaseList.tsx:43]: need to move the search value 1 level up, as it needs to stay when a user navigates between clusters.

**Review by alex-kovoy — COMMENTED** [review]
with a minor comment (move the search input value 1 level up)

**Review by alex-kovoy — APPROVED** [review]

**Pull request merged**